lib/active_model/serializer.rb

```diff
@@ -50,20 +50,16 @@ def self.inherited(base)
 
     def self.attributes(*attrs)
       attrs = attrs.first if attrs.first.class == Array
-      @_attributes.concat attrs
-      @_attributes.uniq!
 
       attrs.each do |attr|
-        define_method attr do
-          object && object.read_attribute_for_serialization(attr)
-        end unless method_defined?(attr) || _fragmented.respond_to?(attr)
+        attribute(attr)
       end
     end
 
     def self.attribute(attr, options = {})
       key = options.fetch(:key, attr)
-      @_attributes_keys[attr] = { key: key } if key != attr
-      @_attributes << key unless @_attributes.include?(key)
+      self._attributes_keys[attr] = { key: key } if key != attr
```

Comment:
so, because `_attributes` is a hash, does that mean it's not ok, because it's being accessed and the `attr` value is being set?

or is it ok, because *something* is getting set using `_attributes`?